src/components/sidebar/CallTreeSidebar.js

```diff
@@ -40,26 +45,76 @@ function SidebarDetail({ label, children }: SidebarDetailProps) {
   );
 }
 
+type ImplementationBreakdownProps = {|
+  +breakdown: BreakdownByImplementation,
+|};
+
+// This component is responsible for displaying the breakdown data specific to
+// the JavaScript engine and native code implementation.
+class ImplementationBreakdown extends React.PureComponent<
+  ImplementationBreakdownProps
+> {
+  _orderedImplementations: $ReadOnlyArray<StackImplementation> = [
+    'native',
+    'interpreter',
+    'baseline',
+    'ion',
+    'unknown',
+  ];
+
+  _labelizeImplementation(implementation: StackImplementation): string {
```

Comment:
As we discussed on slack, I'll keep it this way for now because it's more "object oriented" and we use it only here, but it's easy to extract to a function if we need it elsewhere in the future.